catalog/spatial/registry/registry-common/src/main/java/org/codice/ddf/registry/common/metacard/RegistryUtility.java

```diff
@@ -0,0 +1,127 @@
+     *
+     * @param metacard - the metacard for which the registry Id will be returned
+     * @return the String representation of the metacard's registry id attribute, null if not
+     * present
+     */
+    public static String getRegistryId(Metacard metacard) {
+        return getStringAttribute(metacard, RegistryObjectMetacardType.REGISTRY_ID, null);
+    }
+
+    /**
+     * Identifies whether the metacard is a registry identity node
+     *
+     * @param metacard - metacard to be evaluated for registry identity status
+     * @return true if the metacard is an identity node, false otherwise
+     */
+    public static boolean isIdentityNode(Metacard metacard) {
+        Attribute attribute =
+                metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_IDENTITY_NODE);
+        return attribute != null && (boolean) attribute.getValue();
+    }
+
+    /**
+     * Identifies whether the metacard is a local registry node
+     *
+     * @param metacard - metacard to be evaluated for local registry status
+     * @return true if the metacard is an local node, false otherwise
+     */
+    public static boolean isLocalNode(Metacard metacard) {
+        Attribute attribute =
+                metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_LOCAL_NODE);
+        return attribute != null && (boolean) attribute.getValue();
+    }
+
+    /**
+     * Attempts to return a String attribute's value. If the attribute is null the default value
+     * will be returned instead.
+     *
+     * @param metacard         - metacard to be evaluated for a String type attribute
+     * @param attributeToCheck - attribute to be checked for, should have a String type
+     * @param defaultValue     - the value expected to be returned if the attribute is not present
+     * @return the defaultValue if attribute is not present, otherwise a String representing the
+     * value of the attribute
+     */
+    public static String getStringAttribute(Metacard metacard, String attributeToCheck,
+            String defaultValue) {
+        Attribute attribute = metacard.getAttribute(attributeToCheck);
+        if (attribute == null) {
+            return defaultValue;
+        }
+        return attribute.getValue()
+                .toString();
+    }
+
+    /**
+     * Attempts to return a List of Strings from an attribute. If the attribute is null the default
+     * value will be returned instead.
+     *
+     * @param metacard         - metacard to be evaluated for a List of Strings type attribute
+     * @param attributeToCheck - attribute to be checked for, should be a List of Strings type
+     * @param defaultValue     - the value expected to be returned if the attribute is not present
+     * @return the defaultValue if attribute is not present, otherwise a List of Strings
+     * representing the values of the attribute
+     */
+    public static List<String> getListOfStringAttribute(Metacard metacard, String attributeToCheck,
```

Comment:
I think I used it to return null as well as empty list in a few areas, but will refactor to check for empty list instead of null.